src/com/yetanalytics/lrs_admin_ui/views/browser.cljs

```diff
@@ -2,23 +2,154 @@
-                                     :params (.-search elem)}]))))
+(defn actor-display 
+  "Actor IFI progressive resolution to a display string."
+  [actor]
+  (or (get actor "name")
+      (get-in actor ["account" "name"])
+      (get actor "mbox")
+      (get actor "mbox_sha1sum")
+      (get actor "openid")))
+
+(defn object-display
+  "Object display resolution including the possibility of actor object"
+  [object]
+  (or (langmap (get-in object ["definition" "name"]))
+      (get object "id")
+      (actor-display object)))
+
+(defn verb-display
+  "Verb display resolution"
+  [verb]
+  (or (langmap (get verb "display"))
+      (get verb "id")))
+
+(defn filter-button
+  [{:keys [key value title]}]
+  [:a {:class "pointer"
+       :title title
+       :on-click #(dispatch [:browser/add-filter key value])}
+   [:img {:src "images/icons/add-filter.svg"
+          :width "15px"
+          :height "20px"}]])
+
+(defn verb-cell
+  [row]
+  (let [{:strs [verb]} (js->clj row)
+        {:strs [id]} verb]
+    (r/as-element
+     [:div.custom-cell
+      [:div.cell-display
+       (verb-display verb)]
+      [:div.cell-action 
+       [filter-button {:title "Filter by Verb"
+                       :key   :verb
+                       :value id}]]])))
+
+(defn actor-cell
+  [row]
+  (let [{:strs [actor]} (js->clj row)]
+    (r/as-element
+     [:div.custom-cell
+      [:div.cell-display
+       (actor-display actor)]
+      [:div.cell-action
+       [filter-button {:title "Filter by Actor"
+                       :key   :agent
+                       :value (js/JSON.stringify (clj->js actor))}]]])))
+
+(defn object-cell 
+  [row]
+  (let [{:strs [object]} (js->clj row)
+        {:strs [id]} object]
+    (r/as-element
+     [:div.custom-cell
+      [:div.cell-display
+       (object-display object)]
+      [:div.cell-action
+       (when id
+         [filter-button {:title "Filter by Activity"
+                         :key   :activity
+                         :value id}])]])))
+
+(defn row-num-cell
+  [_ idx]
+  (let [page-start 
+        (+ 1 (* @(subscribe [:browser/get-batch-size])
+                (count @(subscribe [:browser/get-back-stack]))))]
+    (r/as-element [:i (+ idx page-start)])))
+
+(defn view-statement-json
+  [row]
+  (json-viewer
+   {:data (:data row)
+    ;; top level expanded only
+    :collapsed 1}))
+
+(defn statement-table 
+  [{:keys [data]}]
+  (let [cols [{:name (r/as-element [:i "#"])
+               :cell row-num-cell
+               :maxWidth "10px"}
+              {:name "Statement ID"
+               :selector #(get % "id")
+               ;; ensure id is readable
+               :minWidth "300px"}
+              {:name "Actor"
+               :cell actor-cell}
+              {:name "Verb"
+               :cell verb-cell}
+              {:name "Object"
+               :cell object-cell}
+              {:name "Timestamp"
+               :selector #(time/iso8601->local-display (get % "timestamp"))}]
```

Comment:
This could and should be extracted into its own def, no?

Reply:
Could yes. Should why? It's the specific fn for building this specific table, why shouldnt it have the cols?

Reply:
Mainly formatting/reducing clutter tbh. So it's not that important in the grand scheme of things; I just thought it was unusual at first glance.

Reply:
id pull it out if it was reusable or something but its fine where it is since its specific to this instantiation